### PR TITLE
[DEVOPS-1222] Build with iohk/nix and nix-tools.

### PR DESCRIPTION
**Pull request by jbgi**

### CODE REVIEW

File: release.nix [hunk 1]
```diff
@@ -1,24 +1,48 @@
...
+  #
+  #   libs.cardano-chain.x86_64-darwin -- will build the cardano-chain library on and for macOS
+  #   libs.cardano-chain.x86_64-linux -- will build the cardano-chain library on and for linux
+  #   libs.x86_64-pc-mingw32-cardano-chain.x86_64-linux -- will build the cardano-chain library on linux for windows.
+  #   tests.cs-ledger.ledger-delegation-test.x86_64-linux -- will build and run the ledger-delegation-test from the
+  #                                                          cs-ledger package on linux.
+  #
+  required-name = "iohk-monitoring-required-checks";
+  required-targets = jobs: [
+    jobs.nix-tools.exes.iohk-monitoring.example-simple.x86_64-linux
```
Comment (angerman) [inline release.nix]: We'll need to find a better solution for this :(

**Comment by jbgi** [comment]
bors try

**Comment by disassembler** [comment]
bors ping

**Comment by disassembler** [comment]
bors try

**Comment by CodiePP** [comment]
I am trying to build with nix:
```
nix build -f release.nix nix-tools.tests.iohk-monitoring.tests.x86_64-linux
```
but I have obviously to indicate to get the binaries from iohk cache.

and later also for cross-compilation:
```
nix build -f release.nix nix-tools.libs.x86_64-pc-mingw32-iohk-monitoring.x86_64-linux
```

**Comment by jbgi** [comment]
Right, to avoid burning too much cpu, the iohk cache is essential:
https://github.com/input-output-hk/cardano-sl/blob/master/docs/nix.md#binary-cache

the windows build currently fails due to https://github.com/psibi/download/issues/17.

**Comment by jbgi** [comment]
bors try

**Comment by jbgi** [comment]
bors try

**Comment by cleverca22** [comment]
bors try

**Comment by angerman** [comment]
bors try

**Comment by jbgi** [comment]
bors try

**Comment by jbgi** [comment]
bors try

**Comment by jbgi** [comment]
bors try

**Comment by jbgi** [comment]
bors try

**Comment by jbgi** [comment]
Hydra is undergoing a update, need to wait for it to be redeployed to get bors working properly.
Merging manually as this PR has been hanging out for too much time.

**Pull request merged**